presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -246,10 +246,7 @@ public PlanWithProperties visitEnforceSingleRow(EnforceSingleRowNode node, Conte
         @Override
         public PlanWithProperties visitAggregation(AggregationNode node, Context context)
         {
-            HashSet<Symbol> partitioningRequirement = new HashSet<>(node.getGroupingSets().get(0));
-            for (int i = 1; i < node.getGroupingSets().size(); i++) {
-                partitioningRequirement.retainAll(node.getGroupingSets().get(i));
-            }
+            Set<Symbol> partitioningRequirement = ImmutableSet.copyOf(node.getGroupingKeys());
```

Comment:
Sorry, I missed Agg**Partial** :)

I don't see what is the potential problem that you are referring. When `AddExchanges` is being executed, there are no partial aggregations in the plan. They are created and pushed down as much as possible in latter steps of the optimizer. So this line of code that you are commenting should only affect exchanges between SINGLE/FINAL aggregations and whatever children it has (before creating partials).